File: CPP/stacks.cpp
#include<iostream>
using namespace std;

struct Node{
	int data;
	Node *link;
};

class stacks{
	Node *top;
	int len;
public:
	stacks(){
		top = new Node;
		top = NULL;
		len = 0;
	}	

	void push(int val){
		Node *temp = new Node;
		temp->data = val;
		temp->link = top;
		top = temp;
		len++;
	}

	void pop(){
		Node *temp;
		temp = top;

		if(len != 0){
		cout<<"Popped data is "<<temp->data<<endl;
		top = top->link;
		delete temp;
		len--;
		}else{
			cout<<"Stack is empty !!!"<<endl;
		}
	}
};

int main(){
	stacks s;
	s.push(1);
	s.push(2);
	s.push(3);
	s.pop();
	s.pop();
	s.pop();
	s.pop();
	return 0;
}

File: Python/stacks.py
class linked_list:
    class Node:
        def __init__(self, data, next):
            self.data = data
            self.next = next

    def __init__(self):
        self.size = 0
        self.head = None

    def __len__(self):
        return self.size

    def push(self, e):
        self.head = self.Node(e, self.head)
        self.size += 1

    def pop(self):
        if self.size == 0:
            print("Empty List")
        else:
            temp = self.head.data
            self.head = self.head.next
            self.size -= 1
            return temp


l = linked_list()
print(len(l))
l.push(1)
l.push(2)
print(len(l))
print(l.pop())
